pkg/telemetry/phonehome/client_config.go

```diff
@@ -82,3 +97,43 @@ func (cfg *Config) Telemeter() Telemeter {
 	})
 	return cfg.telemeter
 }
+
+// AddInterceptorFunc appends the custom list of telemetry interceptors with the
+// provided function.
+func (cfg *Config) AddInterceptorFunc(event string, f Interceptor) {
+	mux.Lock()
+	defer mux.Unlock()
+	if cfg.interceptors == nil {
+		cfg.interceptors = make(map[string][]Interceptor, 1)
+	}
```

Comment:
There's no c-tor for Config because there would be too many arguments to pass to it. Instead, a client just fills the required fields of the struct. `interceptors` map is not required to be initialized unless it's going to be used.